In [1]:
import pandas as pd

In [2]:
df_tech_sp500=pd.read_csv("../../processed/tech_sp500_stable_extended_tickers.csv")

In [3]:
df_tech_sp500

cik                           name entityType     sic  \
0     320193                     Apple Inc.  operating  3571.0   
1     789019                 MICROSOFT CORP  operating  7372.0   
2    1045810                    NVIDIA CORP  operating  3674.0   
3    1652044                  Alphabet Inc.  operating  7370.0   
4    1326801           Meta Platforms, Inc.  operating  7370.0   
..       ...                            ...        ...     ...   
119    78814          PITNEY BOWES INC /DE/  operating  3579.0   
120  1770450            Xerox Holdings Corp  operating  3577.0   
121    61986               MANITOWOC CO INC  operating  3531.0   
122  1881551       Solidion Technology Inc.  operating  3690.0   
123  1621672  Super League Enterprise, Inc.  operating  7373.0   

                                        sicDescription          ownerOrg  \
0                                 Electronic Computers     06 Technology   
1                        Services-Prepackaged Software     06 Technology   
2                     Semiconductors & Related Devices  04 Manufacturing   
3    Services-Computer Programming, Data Processing...     06 Technology   
4    Services-Computer Programming, Data Processing...     06 Technology   
..                                                 ...               ...   
119                               Office Machines, NEC     06 Technology   
120                 Computer Peripheral Equipment, NEC     06 Technology   
121                     Construction Machinery & Equip     06 Technology   
122  Miscellaneous Electrical Machinery, Equipment ...  04 Manufacturing   
123        Services-Computer Integrated Systems Design     06 Technology   

     insiderTransactionForOwnerExists  insiderTransactionForIssuerExists  \
0                                   0                                  1   
1                                   1                                  1   
2                                   1                                  1   
3                                   1                                  1   
4                                   0                                  1   
..                                ...                                ...   
119                                 0                                  1   
120                                 0                                  1   
121                                 1                                  1   
122                                 0                                  1   
123                                 0                                  1   

        tickers      exchanges  ...  mailing_stateOrCountryDescription  \
0          AAPL         Nasdaq  ...                                 CA   
1          MSFT         Nasdaq  ...                                 WA   
2          NVDA         Nasdaq  ...                                 CA   
3    GOOGL,GOOG  Nasdaq,Nasdaq  ...                                 CA   
4       META,FB         Nasdaq  ...                                 CA   
..          ...            ...  ...                                ...   
119  PBI,PBI-PB      NYSE,NYSE  ...                                 CT   
120         XRX         Nasdaq  ...                                 CT   
121         MTW           NYSE  ...                                 WI   
122         STI         Nasdaq  ...                                 TX   
123         SLE         Nasdaq  ...                                 CA   

               business_street1    business_street2  business_city  \
0            ONE APPLE PARK WAY                 NaN      CUPERTINO   
1             ONE MICROSOFT WAY                 NaN        REDMOND   
2     2788 SAN TOMAS EXPRESSWAY                 NaN    SANTA CLARA   
3     1600 AMPHITHEATRE PARKWAY                 NaN  MOUNTAIN VIEW   
4                    1 META WAY                 NaN     MENLO PARK   
..                          ...                 ...            ...   
119        

In [4]:
df_his_weight=pd.read_csv("../../processed/his_weight_tech_500_12_24_in_progress.csv")

In [5]:
df_his_weight

date   rank ticker                                name  \
0       2012-04-30    1.0  AAPL*                           APPLE INC   
1       2012-04-30    3.0    IBM  INTERNATIONAL BUSINESS MACHINES CO   
2       2012-04-30    4.0   MSFT                      MICROSOFT CORP   
3       2012-04-30    7.0      T                            AT&T INC   
4       2012-04-30   15.0  GOOGL                  GOOGLE INC CLASS A   
...            ...    ...    ...                                 ...   
237258  2024-10-30  485.0    HAS                          HASBRO INC   
237259  2024-10-30  489.0   PAYC                 PAYCOM SOFTWARE INC   
237260  2024-10-30  496.0   QRVO                           QORVO INC   
237261  2024-10-30  500.0   PARA            PARAMOUNT GLOBAL CLASS B   
237262  2024-10-30  501.0    FOX                    FOX CORP CLASS B   

                            sector    weight  
0           Information Technology  4.306649  
1           Information Technology  1.896926  
2           Information Technology  1.890466  
3       Telecommunication Services  1.542570  
4           Information Technology  1.213489  
...                            ...       ...  
237258      Consumer Discretionary  0.017720  
237259                 Industrials  0.017020  
237260      Information Technology  0.014060  
237261               Communication  0.012860  
237262               Communication  0.010430  

[237263 rows x 6 columns]

In [10]:
df_his_weight_14_24=df_his_weight[df_his_weight.date>"2013-12-31"]

In [11]:
df_his_weight_14_24

date   rank ticker                                name  \
26936   2014-01-02    1.0  AAPL*                           APPLE INC   
26937   2014-01-02    3.0  GOOGL                  GOOGLE INC CLASS A   
26938   2014-01-02    4.0   MSFT                      MICROSOFT CORP   
26939   2014-01-02   13.0    IBM  INTERNATIONAL BUSINESS MACHINES CO   
26940   2014-01-02   14.0      T                            AT&T INC   
...            ...    ...    ...                                 ...   
237258  2024-10-30  485.0    HAS                          HASBRO INC   
237259  2024-10-30  489.0   PAYC                 PAYCOM SOFTWARE INC   
237260  2024-10-30  496.0   QRVO                           QORVO INC   
237261  2024-10-30  500.0   PARA            PARAMOUNT GLOBAL CLASS B   
237262  2024-10-30  501.0    FOX                    FOX CORP CLASS B   

                            sector    weight  
26936       Information Technology  3.038308  
26937       Information Technology  1.906967  
26938       Information Technology  1.723382  
26939       Information Technology  1.156141  
26940   Telecommunication Services  1.124044  
...                            ...       ...  
237258      Consumer Discretionary  0.017720  
237259                 Industrials  0.017020  
237260      Information Technology  0.014060  
237261               Communication  0.012860  
237262               Communication  0.010430  

[210327 rows x 6 columns]

In [13]:
len(df_his_weight_14_24.date.unique())

2577

2577 is the number of unique dates between 2014 and 2024 (when there is stock data). 

In [27]:
lst_tics_tech_500_set=[set(str_ele.split(",")) for str_ele in list(df_tech_sp500["extended tickers"])]

In [28]:
lst_tics_tech_500_set

[{'AAPL', 'AAPL*'},
 {'MSFT'},
 {'NVDA'},
 {'GOOG', 'GOOG*', 'GOOGL'},
 {'FB', 'FB*', 'META'},
 {'AVGO', 'BRCM'},
 {'ORCL'},
 {'CRM'},
 {'ADBE'},
 {'AMD'},
 {'TMO'},
 {'TMUS'},
 {'CSCO'},
 {'DHR'},
 {'INTU'},
 {'TXN'},
 {'IBM'},
 {'AMAT'},
 {'NOW'},
 {'VZ'},
 {'CAT'},
 {'CCZ', 'CMCSA'},
 {'T', 'T-PA', 'T-PC', 'TBB', 'TBC'},
 {'MU'},
 {'ETN'},
 {'LRCX'},
 {'ANET'},
 {'ADI'},
 {'PANW'},
 {'MDT'},
 {'ADP'},
 {'DE'},
 {'INTC'},
 {'SNPS'},
 {'APH'},
 {'DELL'},
 {'CDNS'},
 {'TT'},
 {'PLTR'},
 {'ITW'},
 {'NXPI'},
 {'CRWD'},
 {'CARR'},
 {'ROP'},
 {'FTNT'},
 {'ADSK'},
 {'CHTR'},
 {'JCI'},
 {'MPWR'},
 {'MCHP'},
 {'CMI'},
 {'EA'},
 {'VRSK'},
 {'AME'},
 {'CTSH'},
 {'IR'},
 {'SMCI'},
 {'HPQ'},
 {'BKR'},
 {'ON'},
 {'XYL'},
 {'ROK'},
 {'ANSS'},
 {'NTAP'},
 {'VLTO'},
 {'TTWO'},
 {'TYL'},
 {'DOV'},
 {'HPE', 'HPE-PC'},
 {'FTV'},
 {'FSLR'},
 {'KEYS'},
 {'GDDY'},
 {'TER'},
 {'STX'},
 {'WDC'},
 {'PTC'},
 {'HUBB'},
 {'LDOS'},
 {'WBD'},
 {'ZBRA'},
 {'FOX', 'FOXA'},
 {'VRSN'},
 {'SWKS'},
 {'ENPH'},
 {'FDS'},


We leave some room again, we will deem a company is stable if there is less than 7 days of missing data. 

In [29]:
lst_stable_14_24=[]

for ele_set in lst_tics_tech_500_set: 
    lst_stable_14_24.append(len(df_his_weight_14_24[df_his_weight_14_24.ticker.isin(ele_set)])>=2570)

In [30]:
lst_stable_14_24

[True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 False,
 True,
 True,
 True,
 False,
 False,
 True,
 False,
 False,
 False,
 False,
 True,
 False,
 False,
 False,
 True,
 False,
 True,
 False,
 True,
 False,
 True,
 True,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 False,
 False,
 True,
 True,
 False,
 True,
 False,
 False,
 False,
 True,
 False,
 False,
 False,
 False,
 False,
 False,
 True,
 True,
 False,
 False,
 False,
 False,
 False,
 True,
 True,
 False,
 False,
 False,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 False,
 True,
 False,
 False,
 True,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False,
 False]

In [32]:
len(lst_stable_14_24)

124

In [34]:
df_tech_sp500["stable 14 to 24"]=lst_stable_14_24

In [35]:
df_tech_sp500

cik                           name entityType     sic  \
0     320193                     Apple Inc.  operating  3571.0   
1     789019                 MICROSOFT CORP  operating  7372.0   
2    1045810                    NVIDIA CORP  operating  3674.0   
3    1652044                  Alphabet Inc.  operating  7370.0   
4    1326801           Meta Platforms, Inc.  operating  7370.0   
..       ...                            ...        ...     ...   
119    78814          PITNEY BOWES INC /DE/  operating  3579.0   
120  1770450            Xerox Holdings Corp  operating  3577.0   
121    61986               MANITOWOC CO INC  operating  3531.0   
122  1881551       Solidion Technology Inc.  operating  3690.0   
123  1621672  Super League Enterprise, Inc.  operating  7373.0   

                                        sicDescription          ownerOrg  \
0                                 Electronic Computers     06 Technology   
1                        Services-Prepackaged Software     06 Technology   
2                     Semiconductors & Related Devices  04 Manufacturing   
3    Services-Computer Programming, Data Processing...     06 Technology   
4    Services-Computer Programming, Data Processing...     06 Technology   
..                                                 ...               ...   
119                               Office Machines, NEC     06 Technology   
120                 Computer Peripheral Equipment, NEC     06 Technology   
121                     Construction Machinery & Equip     06 Technology   
122  Miscellaneous Electrical Machinery, Equipment ...  04 Manufacturing   
123        Services-Computer Integrated Systems Design     06 Technology   

     insiderTransactionForOwnerExists  insiderTransactionForIssuerExists  \
0                                   0                                  1   
1                                   1                                  1   
2                                   1                                  1   
3                                   1                                  1   
4                                   0                                  1   
..                                ...                                ...   
119                                 0                                  1   
120                                 0                                  1   
121                                 1                                  1   
122                                 0                                  1   
123                                 0                                  1   

        tickers      exchanges  ...            business_street1  \
0          AAPL         Nasdaq  ...          ONE APPLE PARK WAY   
1          MSFT         Nasdaq  ...           ONE MICROSOFT WAY   
2          NVDA         Nasdaq  ...   2788 SAN TOMAS EXPRESSWAY   
3    GOOGL,GOOG  Nasdaq,Nasdaq  ...   1600 AMPHITHEATRE PARKWAY   
4       META,FB         Nasdaq  ...                  1 META WAY   
..          ...            ...  ...                         ...   
119  PBI,PBI-PB      NYSE,NYSE  ...            PITNEY BOWES INC   
120         XRX         Nasdaq  ...               201 MERRITT 7   
121         MTW           NYSE  ...       11270 WEST PARK PLACE   
122         STI         Nasdaq  ...             13355 NOEL ROAD   
123         SLE         Nasdaq  ...  2912 COLORADO AVE, STE 203   

       business_street2  business_city  business_stateOrCountry  \
0                   NaN      CUPERTINO                       CA   
1                   NaN        REDMOND                       WA   
2                   NaN    SANTA CLARA                       CA   
3                   NaN  MOUNTAIN VIEW                       CA   
4                   NaN     MENLO PARK                       CA   
..                  ...            ...                      ...   
119  3001 SUMMER STREET       STAMFORD                       CT   
120                 NaN        NORWALK           

In [37]:
df_tech_sp500.to_csv("../../processed/tech_sp500_stable_extended_tickers.csv")